serial.c

```diff
@@ -64,24 +61,16 @@ void serial_init()
   // defaults to 8-bit, no parity, 1 stop bit
 }
 
-
-
 void serial_sendchar(uint8_t data) 
 {
-  // Calculate next head
-  uint8_t next_head = tx_buffer_head + 1;
-  if (next_head == TX_BUFFER_SIZE) { next_head = 0; }
+  // As this is an interrupt driven UART, we can simply spin forever
+  // and the service routine will drain the queue until there is
+  // enough room
+  while (queue_is_full(&tx_buf));
 
-  // Wait until there is space in the buffer
-  while (next_head == tx_buffer_tail) {
-    if (SYS_EXEC & EXEC_RESET) { return; } // Only check for abort to avoid an endless loop.
-  }
-  // Store data and advance head
-  tx_buffer[tx_buffer_head] = data;
-  tx_buffer_head = next_head;
+  queue_enqueue(&tx_buf, &data);
 
-  // Enable Data Register Empty Interrupt to make sure tx-streaming is running
```

Comment:
Did you intentionally take out the comment? I think there should be at least some comment to explain it the register.